src/librustc_mir/hair/pattern/mod.rs

```diff
@@ -290,15 +290,15 @@ impl<'tcx> fmt::Display for Pattern<'tcx> {
                 write!(f, "{}", subpattern)
             }
             PatternKind::Constant { value } => {
-                fmt_const_val(f, value)
+                write!(f, "{}", &value)
```

Comment:
These `value` fields are `ty::Const` not `&ty::Const`. With the changes you suggested (only lifting entire `&ty::Const`s) I don't see how I can write this code.